src/audio/pipeline/pipeline-schedule.c

```diff
@@ -123,6 +125,12 @@ static enum task_state pipeline_task_cmd(struct pipeline *p,
 			/* No delay: the final stage has already run too */
 			err = SOF_TASK_STATE_RESCHEDULE;
 		} else if (p->status == COMP_STATE_PAUSED) {
+			/* reset the pipeline components for IPC4 after the STOP trigger */
+			if (cmd == COMP_TRIGGER_STOP && IPC4_MOD_ID(host->ipc_config.id)) {
```

Comment:
This is a bit non-obvious that "IPC4_MOD_ID()" is not true in IPC3 builds, but this is used elsewhere already, so this is ok. Making this more obvious something to put in the thinking hat (for all of us).